torch/csrc/distributed/c10d/CUDASymmetricMemoryOps.cu

```diff
@@ -808,14 +810,20 @@ at::Tensor memset32_(
       ")");
 
   auto addr = reinterpret_cast<uint32_t*>(input.data_ptr()) + offset;
-
   c10::cuda::CUDAGuard guard(input.device());
+
+#if !defined(USE_ROCM) && defined(PYTORCH_C10_DRIVER_API_SUPPORTED)
   auto driver_api = c10::cuda::DriverAPI::get();
   C10_CUDA_DRIVER_CHECK(driver_api->cuMemsetD32Async_(
       reinterpret_cast<CUdeviceptr>(addr),
       val,
       count,
       at::cuda::getCurrentCUDAStream()));
+#elif defined(USE_ROCM)
+  C10_HIP_CHECK(hipMemsetD32Async(reinterpret_cast<hipDeviceptr_t>(addr),
+                                   val,
+                                   count,
+                                   at::hip::getCurrentHIPStreamMasqueradingAsCUDA()));
```

Comment:
insteresing, in other places we can use `at::cuda::getCurrentCUDAStream()` without this? Pretty much everywhere kernels are submitted in at::native::cuda

Reply:
O sure.. I can use `at::cuda::getCurrentCUDAStream()` usually that gets hipified into `at::hip::getCurrentHIPStreamMasqueradingAsCUDA()` during build time. Since, this code was already hipified I used the hip version. But, I can use the cuda version also and it will just get hipified later.